src/services/exchange/exchange.service.ts

```diff
@@ -0,0 +1,49 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { HttpErrorHandler } from '../errors/http-error-handler';
+import { ExchangeResult } from './entities/exchange.entity';
+
+@Injectable()
+export class ExchangeService {
+  // TODO can we depend on the base url instead?
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly httpService: HttpService,
+    private readonly httpErrorHandler: HttpErrorHandler,
+  ) {}
+
+  async convertRates(to: string, from: string): Promise<number> {
+    const exchangeResult = await this.getExchangeResult();
+
+    if (exchangeResult.rates === undefined)
+      throw new InternalServerErrorException(`Exchange rates unavailable`);
+
+    const fromExchangeRate = exchangeResult.rates[from];
+    if (fromExchangeRate === undefined)
+      throw new InternalServerErrorException(
+        `Exchange rate for ${from} is not available`,
+      );
+    const toExchangeRate = exchangeResult.rates[to];
+    if (toExchangeRate === undefined)
+      throw new InternalServerErrorException(
+        `Exchange rate for ${to} is not available`,
+      );
+
+    return toExchangeRate / fromExchangeRate;
```

Comment:
This could lead to divisions by zero? I know having a 0 rate value shouldn't happen, do you think is too much defensive programming to consider that case. 

Apart from that, maybe an unit this for this function could help visualize the cases (this is checking for `undefined`, could the `from/to` value be null?) and refactor. (I can do it in a separate PR, no problem 😄)

Reply:
For the test cases I can consider it for https://github.com/5afe/safe-client-gateway-nest/pull/31 (this PR was already big).

For the implementation part I'm going to double check the current implementation regarding this! 👍 Good call. For the implementation we can tackle in this PR.

Ideally I'd have feature+tests here but I was playing a little bit with Faker due to the amount of instances that we were creating for the tests so I decided to do it separately.

Reply:
> For the implementation part I'm going to double check the current implementation regarding this! 👍 Good call. For the implementation we can tackle in this PR.

@hectorgomezv For the current implementation it seems that code might `panic` and return a `500` https://github.com/safe-global/safe-client-gateway/blob/dd162f73150e8de4ea6eda481da1ee802692bdbe/src/providers/fiat.rs#L47

I will add a condition to this to have a better error message from the service.